plugins/plugins.go

```diff
@@ -63,19 +68,86 @@ func (m *Manager) Register(plugin Plugin) {
 	m.plugins = append(m.plugins, plugin)
 }
 
+func (m *Manager) GetCompiler() *ast.Compiler {
+	m.mtx.RLock()
+	defer m.mtx.RUnlock()
+	return m.Compiler
+}
+
+func (m *Manager) setCompiler(compiler *ast.Compiler) {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+	m.Compiler = compiler
+}
+
 // Start starts the manager.
 func (m *Manager) Start(ctx context.Context) error {
 	if m == nil {
 		return nil
 	}
+
+	// open transaction
+	txn, err := m.Store.NewTransaction(ctx, storage.WriteParams)
+	if err != nil {
+		return err
+	}
+
+	// load policies from storage
+	compiler, err := loadCompilerFromStore(m.Store, txn, ctx)
+	if err != nil {
+		m.Store.Abort(ctx, txn)
+		return err
+	}
+
+	m.Store.Commit(ctx, txn)
+
+	// set compiler on manager
+	m.setCompiler(compiler)
+
 	for _, p := range m.plugins {
 		if err := p.Start(ctx); err != nil {
 			return err
 		}
 	}
+
+	storage.Txn(ctx, m.Store, storage.WriteParams, func(txn storage.Transaction) error {
+		m.Store.Register(ctx, txn, storage.TriggerConfig{OnCommit: func(ctx context.Context, txn storage.Transaction, event storage.TriggerEvent) {
+			if event.PolicyChanged() {
+				compiler, _ := loadCompilerFromStore(m.Store, txn, ctx)
+				m.setCompiler(compiler)
+			}
+		}})
+
+		return nil
+	})
+
 	return nil
 }
 
+func loadCompilerFromStore(store storage.Store, txn storage.Transaction, ctx context.Context) (*ast.Compiler, error) {
```

Comment:
Nit: ctx should always be the first argument (this is a go style thing.)

Reply:
Done